### DISCUSSION TITLE
How to use data fetching with store?

### DISCUSSION THREAD

**Discussion by Asxer** [nuxt/nuxt#16275] [Questions]
I see that Nuxt is suggesting strange ways to manage data fetching.

https://nuxt.com/docs/getting-started/data-fetching#isomorphic-fetch-and-fetch

I am finally confused about how to use hybrid rendering with the store. 

1. I don't want to keep data in components because it is a terrible anti-pattern when I have the store. I want to have one source of truth so I want to call an action method of the store. 
2. I don't want to use cut ways to call API like `useFetch` that can not into PUT methods or interceptors to implement basic JWT authorization with refreshing. If I need to use it - I need to create a custom implementation with Axios inside it's better just to use Axios initially. 
3. I've found a way to manage it but it doesn't work stable - it uses a Suspense tag and async calling of a store method using await inside the setup hook of a component. And it looks like the setup method is called twice. I'm getting warnings about different renders on the client and the server side. 

Does anybody know how to do it properly? In Nuxt 2 there was a good implementation of a fetch hook inside the component that is called once per render with moving store data to the client side. Can I have something like that in Nuxt 3?

**Comment by motea927** [comment]
Just do action call inside `useAsyncData` like you do in Nuxt2 `fetch` hook,
and notice that return value is necessary (prevent invoke twice)
```ts
<script setup lang="ts">
await useAsyncData('getProfile', async () => {
  await userStore.getProfile();
  return userStore.ID;
});
</script>
```

**Reply by motea927** [reply]
By the way, we know Nuxt3 provide `$fetch` ([ofetch](https://github.com/unjs/ofetch)),
so you can use `$fetch.create` and `nuxt plugin` to do same thing as axios
